apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -3,33 +3,55 @@ defmodule Aecore.Structures.SignedTx do
   Aecore structure of a signed transaction.
   """
 
-  alias Aecore.Keys.Worker, as: Keys
   alias Aecore.Structures.TxData
   alias Aecore.Structures.SignedTx
+  alias Aewallet.Signing
 
+  @typedoc "Structure of the TxData"
+  @type tx() :: %TxData{}
+
+  @typedoc "Structure of the SignedTx module"
   @type t :: %SignedTx{
-    data: TxData.t(),
+    data: tx(),
     signature: binary()
   }
 
   @doc """
-    Definition of Aecore SignedTx structure
+  Definition of Aecore SignedTx structure
 
   ## Parameters
-     - data: Aecore %TxData{} structure
-     - signature: Signed %TxData{} with the private key of the sender
+     * data: Aecore %TxData{} structure
+     * signature: Signed %TxData{} with the private key of the sender
   """
   defstruct [:data, :signature]
   use ExConstructor
 
-  @spec is_coinbase?(SignedTx.t()) :: boolean()
-  def is_coinbase?(tx) do
-    tx.data.from_acc == nil && tx.signature == nil
+  @spec is_coinbase?(t()) :: boolean()
+  def is_coinbase?(%{data: %{from_acc: key}, signature: signature}) do
+    key == nil && signature == nil
+  end
+
+  @spec is_valid?(t()) :: boolean()
+  def is_valid?(%{data: data, signature: signature}) do
+    data.value >= 0 && data.fee >= 0 &&
+      Signing.verify(:erlang.term_to_binary(data), signature, data.from_acc)
   end
 
-  @spec is_valid?(SignedTx.t()) :: boolean()
-  def is_valid?(tx) do
-    tx.data.value >= 0 && tx.data.fee >= 0 && Keys.verify_tx(tx)
+  @doc """
+  Takes the transaction that needs to be signed
+  and the private key of the sender.
+  Returns a signed tx
+
+  ## Parameters
+     - tx: The transaction data that it's going to be signed
+     - priv_key: The priv key to sign with
+
+  """
+  @spec sign_tx(tx(), binary()) :: {:ok, t()}
+  def sign_tx(tx, priv_key) do
+    signature = Signing.sign(:erlang.term_to_binary(tx), priv_key)
+    {:ok, %SignedTx{data: tx, signature: signature}}
```

Comment:
Maybe this should just return the SignedTx without :ok since there are no checks if this is the desired result.

Reply:
Well, I'll keep it like this and make some verifications, as well in the TxData module, because now when you mention it I can see that it's needed.